Source/Csla.AspNetCore/Blazor/State/SessionIdManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SessionIdManager.cs" company="Marimer LLC">
+//     Copyright (c) Marimer LLC. All rights reserved.
+//     Website: https://cslanet.com
+// </copyright>
+// <summary>Manages the per-user id value</summary>
+//-----------------------------------------------------------------------
+using System;
+using Csla.State;
+using Microsoft.AspNetCore.Http;
+
+#nullable enable
+namespace Csla.Blazor.State
+{
+  /// <summary>
+  /// Manages the per-user id value
+  /// for state management using a
+  /// browser cookie and Guid value.
+  /// </summary>
+  /// <param name="httpContextAccessor"></param>
+  public class SessionIdManager(IHttpContextAccessor httpContextAccessor) : ISessionIdManager
+  {
+    private readonly IHttpContextAccessor HttpContextAccessor = httpContextAccessor;
+
+    /// <summary>
+    /// Gets the per-user id value
+    /// for the current user state.
+    /// </summary>
+    /// <remarks>
+    /// Id is a Guid value. The id value is
+    /// maintained in a browser cookie for the
+    /// current user.
+    /// </remarks>
+    public string? GetSessionId()
```

Comment:
Wouldn't it be better to return `Guid?` here so calling code doesn't need to convert to a Guid everytime?

Reply:
I'm creating the API to be neutral about how someone might implement the id value. I'm using a Guid, but someone might use a primary key from a database table or something - hard to predict what someone might do.